bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+    strategies.flatMap(_.analyze(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable, guideProbe: ValidatableGuideProbe, guideStar: SiderealTarget): Option[AgsAnalysis] =
+    AgsAnalysis.analysis(ctx, mt, guideProbe, guideStar, probeBands)
+
+  override def catalogQueries(ctx: ObsContext, mt: MagnitudeTable): List[CatalogQuery] =
+    strategies.flatMap(_.catalogQueries(ctx, mt))
+
+  override def candidates(ctx: ObsContext, mt: MagnitudeTable): Future[List[(GuideProbe, List[SiderealTarget])]] =
+    Future.traverse(strategies)(_.candidates(ctx, mt)).map(_.flatten)
+
+  override def estimate(ctx: ObsContext, mt: MagnitudeTable): Future[AgsStrategy.Estimate] =
+    Future.fold(strategies.map(_.estimate(ctx, mt)))(1.0)((p,est) => p * est.probability).map(Estimate(_))
+
+  // This is a bit hacked, as it requires invoking select for each of the substrategies, which are structured to
+  // pick the best assignment for the current position angle constraint. This, of course, might result in different
+  // position angles for substrategies if we naively call select on each, so instead, we hack the context to
+  // handle each position angle constraint and then pick the best one.
+  override def select(ctx: ObsContext, mt: MagnitudeTable): Future[Option[AgsStrategy.Selection]] = {
+    // We hack the obs context to be a list of obs contexts with all possible position angles, and with fixed
+    // position angle constraint so that guide probes will not be allowed to take on different angles in select calls.
+    def ctxToFixedList: List[ObsContext] = {
+      val ctxFixed = {
+        val instFixed = ctx.getInstrument match {
+          case pac : SPInstObsComp with PosAngleConstraintAware =>
+            val newPac = pac.clone.asInstanceOf[SPInstObsComp with PosAngleConstraintAware]
+            newPac.setPosAngleConstraint(PosAngleConstraint.FIXED)
+            newPac
+          case oth : SPInstObsComp => oth
+        }
+        ctx.withInstrument(instFixed)
+      }
+      (ctx.getPosAngleConstraint match {
+        case PosAngleConstraint.FIXED                                            => List(0)
+        case PosAngleConstraint.FIXED_180 | PosAngleConstraint.PARALLACTIC_ANGLE => List(0, 180)
+        case PosAngleConstraint.UNBOUNDED                                        => Range(0, 360, 15).toList
+      }).map(a => ctxFixed.withPositionAngle(ctx.getPositionAngle.add(a, Angle.Unit.DEGREES)))
+    }
+
+    // For each ObsContext, perform the selection and return it as a future.
+    val noStrategy: Option[AgsStrategy.Selection] = None
+    def ctxSelect(cCtx: ObsContext): Future[(ObsContext, Option[AgsStrategy.Selection])] =
+      Future.fold(strategies.map(_.select(cCtx, mt)))(cCtx -> noStrategy) {
+        case ((_, resOpt), curOpt) =>
+          cCtx -> curOpt.map(cur => AgsStrategy.Selection(cur.posAngle, resOpt.fold(cur.assignments)(cur.assignments ++ _.assignments)))
+      }
```

Comment:
That is a very good observation. I'd think the probes will interfere with each other if using the same star. Perhaps in the particular case of oiwfs and altair it doesn't happen?